modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/view_file_entry_preview_error.jsp

```diff
@@ -23,7 +23,16 @@ FileVersion fileVersion = (FileVersion)request.getAttribute(WebKeys.DOCUMENT_LIB
 			<h3><liferay-ui:message key="no-preview-available" /></h3>
 
 			<p class="text-secondary">
-				<liferay-ui:message arguments="<%= LanguageUtil.formatStorageSize(dlPreviewSizeException.getMaxFileSize(), locale) %>" key="this-file-is-too-large-to-preview.-the-maximum-size-for-preview-files-is-x" />
+
+				<%
+				long maxFileSize = dlPreviewSizeException.getMaxFileSize();
+
+				if (maxFileSize == 0) {
+					maxFileSize = DLProcessorRegistryUtil.getPreviewableProcessorMaxSize();
+				}
+				%>
```

Comment:
Why did you move this logic here instead of passing the `DLProcessorRegistryUtil.getPreviewableProcessorMaxSize()` to the constructor of the Exception, this is in files `*PreviewRendererProvider`? If you prefer this way then you can simply remove the `maxFileSize` attribute from the `DLPreviewSizeException` class, it becomes useless.

Reply:
Also, take into account that the configuration attribute `previewableProcessorMaxSize` will be configurable at group level. The group level can be easily taken from the `fileVersion` in `*PreviewRendererProvider` files. Here I suppose it's harder to know for which `fileVersion` was the exception thrown.

Reply:
> Why did you move this logic here instead of passing the `DLProcessorRegistryUtil.getPreviewableProcessorMaxSize()` to the constructor of the Exception, this is in files `*PreviewRendererProvider`? If you prefer this way then you can simply remove the `maxFileSize` attribute from the `DLPreviewSizeException` class, it becomes useless.

We cannot assume that the exception may have the field with data, so we must have a way to display the value if it does not come.
And to say "if not comes it's a developer problem" it's not good for me. So I prefer to prevent the misuse of the exception that to show bad information.

Reply:
> Also, take into account that the configuration attribute `previewableProcessorMaxSize` will be configurable at group level. The group level can be easily taken from the `fileVersion` in `*PreviewRendererProvider` files. Here I suppose it's harder to know for which `fileVersion` was the exception thrown.

But now we do not have this field so I can not make the method right now with the fileVersion.
And YES we know which fileversion is failing (see the jsp, has fileVersion.getFileEntry()) so in the future we can pass the group of whatever we need. :)

Reply:
Then, please remove the `maxFileSize` attribute from the Exception class. It's now useless.

Reply:
As we talked yesterday, when the PRs about the videos, images, audios and documents unification (now on review)
we will pass the value on the exception for my changes.
But we can not prevent future developers that will call a simply `new DLPreviewSizeException()` and I want to prevent the system present the user wrong information

Reply:
I didn't get that yesterday, sorry.